engines/griffon/griffon.h

```diff
@@ -0,0 +1,551 @@
+	void drawAnims(int Layer);
+	void drawHud();
+	void drawNPCs(int mode);
+	void drawOver(int modx, int mody);
+	void drawPlayer();
+	void drawView();
+	void swash();
+
+	// engine.cpp
+	float RND();
+
+	void mainLoop();
+	void updateEngine();
+	void newGame();
+
+	// gfx.cpp
+	void addFloatIcon(int ico, float xloc, float yloc);
+	void addFloatText(const char *stri, float xloc, float yloc, int col);
+	void eventText(const char *stri);
+	void drawLine(Graphics::TransparentSurface *buffer, int x1, int y1, int x2, int y2, int col);
+	void drawString(Graphics::TransparentSurface *buffer, const char *stri, int xloc, int yloc, int col);
+	void drawProgress(int w, int wm);
+
+	// input.cpp
+	void checkInputs();
+	void handleWalking();
+	void checkTrigger();
+	void processTrigger(int trignum);
+
+	// logic.cpp
+	void updateAnims();
+	void updateY();
+	void updateNPCs();
+	void updateSpells();
+	void updateSpellsUnder();
+
+	// resources.cpp
+	void initialize();
+	Graphics::TransparentSurface *loadImage(const char *name, bool colorkey = false);
+	void loadMap(int mapnum);
+	void loadAnims();
+	void loadFont();
+	void loadItemImgs();
+	void loadTiles();
+	void loadTriggers();
+	void loadObjectDB();
+
+	// saveload.cpp
+	Common::String makeSaveGameName(int slot);
+	int loadState(int slotnum);
+	int loadPlayer(int slotnum);
+	int saveState(int slotnum);
+
+	// sound.cpp
+	void setChannelVolume(int channel, int volume);
+	int getSoundHandle();
+	int playSound(DataChunk *chunk, bool looped = false);
+	void pauseSoundChannel(int channel);
+	void haltSoundChannel(int channel);
+	void resumeSoundChannel(int channel);
+	bool isSoundChannelPlaying(int channel);
+	void setupAudio();
+	void updateMusic();
+
+private:
+	Graphics::TransparentSurface *_video, *_videoBuffer, *_videoBuffer2, *_videoBuffer3;
+
+	// system
+	Graphics::TransparentSurface *_titleImg, *_titleImg2, *_inventoryImg;
+	Graphics::TransparentSurface *_logosImg, *_theEndImg;
+	Common::Event _event;
+
+	Graphics::TransparentSurface *_mapBg, *_clipBg, *_clipBg2;
+	unsigned int _clipSurround[4][4];
+
+	float _animSpeed; // CHECKME: it seems to always be 0.5
+	int _rampData[40][24];
+
+	int _curMap;
+	Graphics::TransparentSurface *_fontChr[224][5]; // 256 - 32
+	Graphics::TransparentSurface *_itemImg[21], *_windowImg;
+	Graphics::TransparentSurface *_spellImg;
+
+	bool _itemSelOn;
+	int _curItem, _itemTicks;
+	float _itemyloc;
+	bool _selEnemyOn;
+	int _curEnemy;
+	bool _forcePause;
+	bool _roomLock; // set to disable any room jumps while in the room
+	int _scriptFlag[100][10], _saveSlot;  // script, flag
+
+	// timer related - move to local later
+	int _ticks, _ticksPassed, _nextTicks;
+	float _fp, _fps, _fpsr; // CHECKME: _fp and _fps seems to be integers
+	int _secsingame, _secStart;
+
+	Graphics::TransparentSurface *mapImg[4];
+
+	Common::Rect rcSrc, rcDest;
+
+	// -----------special case
+	bool _dontDrawOver;   // used in map24 so that the candles don't draw over the boss, default set to 0
```

Comment:
Nitpick, but presumably default is set to "false"

Reply:
Fixed